docs/examples/video_decoding/README.md

```diff
@@ -0,0 +1,42 @@
+# Decoding video in DALI backend
+
+## Splitting sequences into subsequences
+Often the video model we want to run inference on expects sequences of some fixed lentght. In such case we may want to split the decoded video
+into a batch of sub-sequences of that given length. We can achieve that utilizing the `fn.reshape` operator from DALI together with a model parameter
+`"split_along_outer_axis"` set in configuration file. Lets see an example of using this combination.
+
+```python
+FRAMES_PER_SEQUENCE = 5
+OUT_WIDTH = 300
+OUT_HEIGHT = 300
+
+@autoserialize
+@dali.pipeline_def(batch_size=256, num_threads=4, device_id=0,
+                   output_dtype=dali.types.UINT8, output_ndim=5)
+def pipeline():
+  vid = fn.external_source(device='cpu', name='INPUT', ndim=1, dtype=dali.types.UINT8)
+  seq = fn.experimental.decoders.video(vid, device='mixed')
+  seq = fn.resize(seq, resize_x=OUT_WIDTH, resize_y=OUT_HEIGHT)
+
+  seq = fn.pad(seq, axis_names='F', align=FRAMES_PER_SEQUENCE)
+  return fn.reshape(seq, shape=[-1, FRAMES_PER_SEQUENCE, OUT_HEIGHT, OUT_WIDTH, 3], name='OUTPUT')
+```
+You can see that we added two operators at the end of the pipeline - `pad` and `reshape`. Padding samples is needed because we want to split them into sub-sequneces of `FRAMES_PER_SEQUENCE` frames each - that means that we need to ensure that the total number of frames in a sample is divisible by `FRAMES_PER_SEQUENCE`. The result of this operator is adding frames filled with zeroes to the end of the sample to align to that number.
+
+After padding the samples, we use the `reshape` operator to add another dimension to the data, so layout-wise the transformation could be represented as `FHWC` -> `NFHWC`, where `N` dimension represents index of a sub-sequence. An example flow could be described as follows: lets say a sample after resizing has a shape of `(13, 300, 300, 3)` - it's a sequence of 13 frames. Assuming the `FRAMES_PER_SEQUENCE` is 5, the sample is then padded to shape `(15, 300, 300, 3)`. Then, the reshape, without modifying any data, changes it to `(3, 5, 300, 300, 3)` - 3 sub-sequences with 5 frames each.
```

Comment:
How about expressing this transformation using the example shapes in a pipeline. I mean something like:
```python
def pipeline():
  vid = fn.external_source(device='cpu', name='INPUT', ndim=1, dtype=dali.types.UINT8)  # (6, 2859274)
  seq = fn.experimental.decoders.video(vid, device='mixed')  # (6, 300, 300, 3)
  seq = fn.pad(seq, axis_names='F', align=FRAMES_PER_SEQUENCE)  # (10, 300, 300, 3)
  return fn.reshape(seq, shape=[-1, FRAMES_PER_SEQUENCE, OUT_HEIGHT, OUT_WIDTH, 3], name='OUTPUT')  # (2, 5, 300, 300, 3)
```
This might be attached to the example above, or maybe specified in some pseudo code, or a chain of operations? Like `input(6, 2859274) -> decode(6, 300, 300, 3)` etc...

The point here would be to guide the user what are the shape changes across the pipeline.

Reply:
Alternatively, you could specify a concrete "problem" at the top of this document and follow this example throughout. Something for example:
```
Let's say we have a 3 video files, which have different number of frames: 20, 13 and 18. The model we want to use always takes a sequence of 4 frames as an input
```
That's just a suggestion, it may make this description better understandable.

Reply:
I did it as python snippet interleaved with comments describing the changes of sample shape